packages/flutter/lib/src/foundation/change_notifier.dart

```diff
@@ -210,7 +243,8 @@ class ChangeNotifier implements Listenable {
   @mustCallSuper
   void dispose() {
     assert(_debugAssertNotDisposed());
-    _listeners = null;
+    _listeners = List<VoidCallback?>.filled(0, null);
+    _disposed = true;
```

Comment:
please set _disposed in an assert, and then rename it _debugDisposed, to indicate that the intent is not to allocate any memory for this flag in release builds.